bqskit/compiler/machine.py

```diff
@@ -1,50 +1,51 @@
-"""
-This module implements the MachineModel class.
-
-A MachineModel models a quantum processing unit's connectivity.
-"""
+"""This module implements the MachineModel class."""
 from __future__ import annotations
 
 import itertools as it
-from typing import Sequence
+from functools import lru_cache
+from typing import Iterable
 
+from bqskit.ir.location import CircuitLocation
+from bqskit.ir.location import CircuitLocationLike
+from bqskit.utils.typing import is_integer
 from bqskit.utils.typing import is_valid_coupling_graph
-from bqskit.utils.typing import is_valid_location
 
 
 class MachineModel:
-    """The MachineModel Class."""
+    """
+    The MachineModel Class.
+
+    A MachineModel models a quantum processing unit's connectivity.
+    """
 
     def __init__(
         self,
         num_qudits: int,
-        coupling_graph: set[tuple[int, int]] | None = None,
+        coupling_graph: Iterable[tuple[int, int]] | None = None,
```

Comment:
I think coupling graphs should support indexing, so they should be sequences.

Reply:
This is just what the user can pass in. It also isn't clear to me why the coupling graphs should support indexing. We are representing them as an edge set. Did you have a use case in mind?